src/SpineAnimation.ts

```diff
@@ -1,31 +1,35 @@
-import { Skeleton } from "./spine-core/Skeleton";
-import { SkeletonData } from "./spine-core/SkeletonData";
-import { AnimationState } from "./spine-core/AnimationState";
-import { AnimationStateData } from "./spine-core/AnimationStateData";
-import { MeshGenerator } from "./core/MeshGenerator";
+import { Skeleton, SkeletonData, AnimationState, AnimationStateData, Physics, Vector2 } from "@esotericsoftware/spine-core";
+import { SpineGenerator } from "./core/SpineGenerator";
 import { SpineRenderSetting } from "./types";
-import { Vector2 } from "./spine-core/Utils";
 import {
-  Script,
+  Renderer,
   Entity,
   ignoreClone,
-  MeshRenderer,
   Texture2D,
   Material,
   Engine,
+  BoundingBox,
+  ShaderMacro,
+  Primitive,
+  Logger,
 } from "@galacean/engine";
 import { SpineMaterial } from "./SpineMaterial";
+import { SubPrimitive } from "./core/SpinePrimitive";
 
-export class SpineAnimation extends Script {
-  /** Spine 材质 */
+export class SpineAnimation extends Renderer {
```

Comment:
This class mixes `renderer` and `animation`, when we want to implement `Spine` in the `UI`, we need to copy the logic of `Animation`.

Reply:
Regarding of this case, I had a discussion with Chen Mo and Cheng Kong before. 
Chen Mo's opinion is whether the UI layer rendering can be handled by engine. Engine's rendering does not distinguish between UI and 3D.
 If there is any compatibility code, it will be handled in SpineGenerator.